### DISCUSSION TITLE
Notifications for @mentions even when not a member of a channel

### DISCUSSION THREAD

**Discussion by chenilim** [mattermost/community#68] [General Ideas]
There is probably a lot of history around this, but it is a common frustration to have to add people to channels when mentioning them. It seems that they should be notified even if not a member, perhaps emailed the thread, and given an option to join and view. At least for a public channel.

It is clunky to force the author to perform those steps, because the intent is to always notify them by the @ mention.

Potential improvement: Maybe there's a way for mentionees to respond to just that thread without joining the channel. In many cases, they need to join, respond, then leave. But what if they get mentioned again?

Even bigger picture: maybe explicitly requiring join / leaves on public channels adds friction? Perhaps just assume everyone is a "member" of all public channels, but allow them to only show their "favorites" on their sidebar. So @ mentions always work, and the un-favorited channels only appear when you have an active mention.

**Comment by esethna** [comment]
Agree, this is a really common use case we should try to optimize

**Comment by amyblais** [comment]
I'm definitely interested in this. Also from my perspective, it would be nice to see if someone mentioned me in a channel that I'm not a member of.

**Comment by Willyfrog** [comment]
In DMs I usually mention people, not to have them pinged but to make sure the other user knows who am I talking about without misleading, maybe we should have a way of not mentioning users but using their handle for ID purposes? (i'm thinking something like !@guillermo.vaya  that would autocomplete, have my user popover, etc. but not trigger notification)
